src/lib/viewers/media/MediaBaseViewer.js

```diff
@@ -60,6 +63,49 @@ class MediaBaseViewer extends BaseViewer {
         this.loadTimeout = 100000;
         this.oldVolume = DEFAULT_VOLUME;
         this.pauseListener = null;
+
+        this.startTimeInSeconds = this.getStartTimeInSeconds();
+    }
+
+    /**
+     * Converts a value and unit to seconds
+     *
+     * @param {string|number} value - the value e.g. 1
+     * @param {string} unit - the unit e.g. seconds
+     * @return {number} a time in seconds
+     */
+    getStartTimeInSeconds() {
+        let convertedValue = INITIAL_TIME_IN_SECONDS;
+
+        const value = getProp(this.fileStartObj, 'value');
+        const unit = getProp(this.fileStartObj, 'unit');
+
+        if (!value || !unit) {
+            return INITIAL_TIME_IN_SECONDS;
+        }
+
+        if (unit === SECONDS_UNIT_NAME) {
+            convertedValue = parseInt(value, 10);
+            if (convertedValue < 0) {
+                // Negative values aren't allowed, start from beginning
+                return INITIAL_TIME_IN_SECONDS;
+            }
+        } else {
+            console.error('Invalid unit for start:', unit); // eslint-disable-line no-console
+        }
+
+        return convertedValue;
+    }
+
+    /**
+     * Takes the start parameter, validates, converts, and applies to the media element
+     *
+     * @return {void}
+     */
+    setStartTime() {
+        if (this.mediaEl && this.startTimeInSeconds > 0 && this.startTimeInSeconds < this.mediaEl.duration) {
```

Comment:
I would prefer that this function acts as a traditional mutator, and accepts a parameter. Wherever you call it and require `this.startTimeInSeconds`, just pass it in. It'll be more flexible that way, as it seems strange to have to set the value of `startTimeInSeconds`, and then additionally call `setStartTime()`.

Another note, instead of calling this `setStartTime()`, would it be better to just do the validation inside of `setMediaTime()` and call it, with `this.startTimeInSeconds`?

Reply:
+1, that way, you mirror what's done in the doc viewer (validation in `setPage()`, pass in `this.startPageNum`)

Reply:
Yeah, I guess this method can be totally removed and the logic combined in ```setMediaTime```